### PR TITLE
feat: update libp2p

### PR DESCRIPTION
**Pull request by simone-sanfratello**
I'm trying to update `libp2p` and related libs, but I'm stuck.
Can anyone help me? @alanshaw ? @vasco-santos ?

### CODE REVIEW

File: test/utils/helpers.js [hunk 1]
```diff
@@ -1,34 +1,30 @@
...
+  const node = await createLibp2p({
+    transports: [new WebSockets()],
+    streamMuxers: [new Mplex()],
+    connectionEncryption: [new Noise()]
+    // TODO connectionEncryption: [new Noise(null, null, noiseCrypto)]
   })
 
-  const connection = await node.dial(`/ip4/127.0.0.1/tcp/${port}/ws/p2p/${peerId}`)
+  const connection = await node.dial(`/ip4/127.0.0.1/tcp/${port}/ws/p2p/${peerId.toString()}`)
   const { stream } = await connection.newStream(protocol)
```
Comment (simone-sanfratello) [inline test/utils/helpers.js:28]: doesn't work, I tried to debug without luck

File: package.json [hunk 2]
```diff
@@ -15,14 +15,17 @@
...
+    "@libp2p/websockets": "^3.0.2",
     "bl": "^5.0.0",
     "dotenv": "^16.0.1",
     "hdr-histogram-js": "^3.0.0",
     "it-length-prefixed": "^6.0.1",
     "it-pipe": "^2.0.4",
     "js-yaml": "^4.1.0",
-    "libp2p": "^0.34.0",
+    "libp2p": "^0.37.3",
     "libp2p-mplex": "^0.10.7",
```
Comment (simone-sanfratello) [inline package.json:29]: to be removed

File: src/peer-id.js [hunk 3]
```diff
@@ -20,15 +20,17 @@ async function downloadPeerIdFile(dispatcher) {
 }
 
 async function getPeerId(dispatcher = defaultDispatcher) {
+  const { createFromJSON, createEd25519PeerId } = await loadEsmModule('@libp2p/peer-id-factory')
```
Comment (simone-sanfratello) [inline src/peer-id.js:23]: I'm not sure this is right, I took from examples and `libp2p` code

File: src/service.js [hunk 4]
```diff
@@ -230,22 +229,22 @@ async function startService({ peerId, currentPort, dispatcher, announceAddr } =
...
+      streamMuxers: [new Mplex()],
+      connectionEncryption: [new Noise()]
+      // TODO connectionEncryption: [new Noise(null, null, noiseCrypto)]
     })
 
-    service.on('error', err => {
-      logger.warn({ err }, `libp2p error: ${serializeError(err)}`)
-    })
+    // TODO?
+    // service.on('error', err => {
```
Comment (simone-sanfratello) [inline src/service.js:245]: Looks like `on` method is not available anymore

Comment (vasco-santos) [reply src/service.js:245]: you can do the following now https://github.com/libp2p/js-libp2p/blob/master/doc/migrations/v0.36-v.037.md#events

File: src/peer-id.js (hunk 3 above)
Comment (vasco-santos) [reply src/peer-id.js:23]: I would say so

File: test/utils/helpers.js (hunk 1 above)
Comment (vasco-santos) [reply test/utils/helpers.js:28]: you probably need to look into updating all other libp2p deps

Comment (vasco-santos) [reply test/utils/helpers.js:28]: https://github.com/libp2p/js-libp2p/blob/master/package.json#L170-L188
Like websockets, mplex, noise. They likely got all braking changes that need to be updated together

**Pull request closed without merging**